x-pack/plugins/maps/public/classes/sources/mvt_single_layer_vector_source/mvt_single_layer_source_settings.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
+import React, { Fragment, Component, ChangeEvent } from 'react';
+import { EuiFieldText, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import _ from 'lodash';
+import { MAX_ZOOM, MIN_ZOOM } from '../../../../common/constants';
+import { ValidatedDualRange, Value } from '../../../../../../../src/plugins/kibana_react/public';
+import { MVTFieldConfigEditor } from './mvt_field_config_editor';
+import { MVTFieldDescriptor } from '../../../../common/descriptor_types';
+
+export type MVTSettings = {
+  layerName: string;
+  fields: MVTFieldDescriptor[];
+  minSourceZoom: number;
+  maxSourceZoom: number;
+};
+
+export interface State {
+  currentSettings: MVTSettings;
+  prevSettings: MVTSettings;
+}
+
+export interface Props {
+  handleChange: (args: MVTSettings) => void;
+  layerName: string;
+  fields: MVTFieldDescriptor[];
+  minSourceZoom: number;
+  maxSourceZoom: number;
+  includeFields: boolean;
+}
+
+export class MVTSingleLayerSourceSettings extends Component<Props, State> {
```

Comment:
I'm getting a warning in the browser about initializing the state properly in `mvt_single_layer_source_settings.tsx`.

![image](https://user-images.githubusercontent.com/1728976/85880880-fc5c2d00-b799-11ea-8e73-f189727dcec6.png)


You can get rid of it by adding a barebones constructor here to init the state:

```
  constructor(props) {
    super(props);
    this.state = {
      prevSettings: null,
      currentSettings: null,
    };
  }

```

Reply:
yes, initializing the settings fixed it